daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/EvTextNumber.scala

```diff
@@ -51,6 +66,163 @@ class TextStandardExponentRepEv(expr: CompiledExpression[String], tci: DPathComp
+    exponentRep: Maybe[String]): DecimalFormat = {
+
+    val dfs = new DecimalFormatSymbols()
+
+    if (decimalSep.isDefined) {
+      dfs.setDecimalSeparator(decimalSep.get)
+    }
+
+    if (groupingSep.isDefined) {
+      dfs.setGroupingSeparator(groupingSep.get)
+    }
+
+    // TODO: this is allowed to be case insenstive, ICU doesn't support that
+    if (exponentRep.isDefined) {
+      dfs.setExponentSeparator(exponentRep.get)
+    }
+
+    if (infRep.isDefined) {
+      // TODO: this is allowed to be case insensitive, ICU doesn't support that
+      dfs.setInfinity(infRep.get)
+    }
+
+    if (nanRep.isDefined) {
+      // TODO: this is allowed to be case insensitive, ICU doesn't support that
+      dfs.setNaN(nanRep.get)
+    }
+
+    val df = new DecimalFormat(textNumberPattern, dfs)
+
+    val cp = checkPolicy match {
+      case TextNumberCheckPolicy.Strict => true
+      case TextNumberCheckPolicy.Lax => false
+    }
+    df.setParseStrict(cp)
+
+    rounding match {
+      case TextNumberRounding.Pattern => {
+        df.setRoundingMode(RoundingMode.HALF_EVEN.ordinal())
+      }
+      case TextNumberRounding.Explicit => {
+        val rm = roundingMode.get match {
+          case TextNumberRoundingMode.RoundCeiling => RoundingMode.CEILING
+          case TextNumberRoundingMode.RoundFloor => RoundingMode.FLOOR
+          case TextNumberRoundingMode.RoundDown => RoundingMode.DOWN
+          case TextNumberRoundingMode.RoundUp => RoundingMode.UP
+          case TextNumberRoundingMode.RoundHalfEven => RoundingMode.HALF_EVEN
+          case TextNumberRoundingMode.RoundHalfDown => RoundingMode.HALF_DOWN
+          case TextNumberRoundingMode.RoundHalfUp => RoundingMode.HALF_UP
+          case TextNumberRoundingMode.RoundUnnecessary => RoundingMode.UNNECESSARY
+        }
+        df.setRoundingMode(rm.ordinal())
+        df.setRoundingIncrement(roundingIncrement.get)
+      }
+    }
+
+    if (isInt) {
+      df.setMaximumFractionDigits(0)
+      df.setDecimalSeparatorAlwaysShown(false)
+      df.setParseIntegerOnly(true)
+    }
+
+    df
+  }
+
+  override protected def compute(state: ParseOrUnparseState): ThreadLocal[DecimalFormat] = {
+
+    val decimalSepList = if (decimalSepEv.isDefined) {
+      val seps = decimalSepEv.get.evaluate(state)
+      if (seps.length > 1) {
+        // TODO: ICU only supports a single decimal separator
+        tci.notYetImplemented("More than one textStandardDecimalSeparator")
+      }
+      MaybeChar(seps.head(0))
+    } else {
+      MaybeChar.Nope
+    }
+
+    val groupingSep = if (groupingSepEv.isDefined) {
+      MaybeChar(groupingSepEv.get.evaluate(state)(0))
+    } else {
+      MaybeChar.Nope
+    }
+
+    val exponentRep = if (exponentRepEv.isDefined) {
+      One(exponentRepEv.get.evaluate(state))
+    } else {
+      Nope
+    }
+
+    checkUnique(
+      decimalSepList,
+      groupingSep,
+      exponentRep)
+
+    val numFormat = new ThreadLocal[DecimalFormat] with Serializable {
```

Comment:
This right here is entirely weird to me. Why would you allocate a thread-local here, when state is passed to this method. 

Can't we just access a method on state here that constructs unique decimal formats for this thread.